### DISCUSSION TITLE
where do i have to put `warnOnPotentiallyInsecureHostPattern: false`

### DISCUSSION THREAD

**Discussion by smoebody** [eclipse-theia/theia#9823] [Q&A]
The documentation on [npm](https://www.npmjs.com/package/@theia/plugin-ext) states, that i have to set `warnOnPotentiallyInsecureHostPattern: false` to suppress the warning about potentially insecure domain in the 'appropriate application configurations' in `package.json`.
Could you be more precise, i.e. add an example?

**Comment by msujew** [comment]
Hi @smoebody thank you for your question.

You can set theia specific application configuration in your `package.json` like this:

```
"theia": {
  "frontend": {
    "config": {
      "warnOnPotentiallyInsecureHostPattern": false
    }
  },
  "backend": {
    "config": {
      "warnOnPotentiallyInsecureHostPattern": false
    }
  }
},
```

This specific setting has to be set for both the frontend as well as the backend config. For a full example how to set the config, see the example [browser app](https://github.com/eclipse-theia/theia/blob/master/examples/browser/package.json) or [electron app](https://github.com/eclipse-theia/theia/blob/master/examples/electron/package.json).

**Reply by vince-fugnitto** [reply]
@smoebody please note that turning off the preference is the unsafe fix, the proper fix is to handle it securely:

- https://github.com/eclipse-theia/theia/tree/master/packages/mini-browser#security-warnings
- https://github.com/eclipse-theia/theia/tree/master/packages/plugin-ext#security-warnings
